staging/src/k8s.io/apimachinery/pkg/apis/meta/internalversion/validation/validation.go

```diff
@@ -44,6 +44,9 @@ func ValidateListOptions(options *internalversion.ListOptions, isWatchListFeatur
 			allErrs = append(allErrs, field.Forbidden(field.NewPath("resourceVersionMatch"), "resourceVersionMatch \"exact\" is forbidden for resourceVersion \"0\""))
 		}
 	}
+	if len(options.Continue) > 0 && len(options.ResourceVersion) > 0 && options.ResourceVersion != "0" {
```

Comment:
it looks like ValidateListOptions is called from four spots: DeleteCollection, ListResource, twice from PrepareWatchListOptionsFromListOptions

does this "continue + non-zero resourceversion" combination currently error in all four paths?

Reply:
Don't know `DeleteCollection` nor `PrepareWatchListOptionsFromListOptions` and I'm little terrified that all those cases are combined. Will followup to verify them.

As for `ListResource` all those requests will be sent to etcd as watchcache doesn't handle any requests setting `continue`. That's why I noticed that this during snapshottable cache implementation as it watchcache didn't validate this.

Reply:
> does this "continue + non-zero resourceversion" combination currently error in all four paths with all other options (watchcache or not, etc)?

No, in other paths (DELETE, WATCH) it doesn't error. The reason is "continue" in that methods is not expected nor used. I think the question is why we are using ListOptions in those paths as they are not using any arguments from them.

Reply:
I'm thinking https://github.com/kubernetes/kubernetes/pull/130637 might be a better direction.

Reply:
> I think the question is why we are using ListOptions in those paths as they are not using any arguments from them.

Not sure what you mean... we do use them:

DeleteCollection does pass listOptions on to List:
https://github.com/kubernetes/kubernetes/blob/4468565250c940bbf70c2bad07f2aad387454be1/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L1313

Watch pulls specific fields out of listOptions (LabelSelector, FieldSelector, AllowWatchBookmarks, SendInitialEvents) and ignores `Continue`: https://github.com/kubernetes/kubernetes/blob/4468565250c940bbf70c2bad07f2aad387454be1/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L1414-L1431